src/d3d9/d3d9_device.cpp

```diff
@@ -1264,8 +1264,9 @@ namespace dxvk {
       // - both destination and source are depth stencil surfaces
       // - both destination and source are offscreen plain surfaces.
       // The only way to get a surface with resource type D3DRTYPE_SURFACE without USAGE_RT or USAGE_DS is CreateOffscreenPlainSurface.
-      if (unlikely(!dstHasRTUsage && (!dstIsSurface || !srcIsSurface || srcHasRTUsage)))
+      if (unlikely((!dstHasRTUsage && (!dstIsSurface || !srcIsSurface || srcHasRTUsage)) && StretchRectValidation)) {
```

Comment:
Maybe just !m_isD3D8Compatible here instead of StretchRectValidation, and avoid the entire wrapper mess? Same for the rest of the disabled validations.